src/lib.rs

```diff
@@ -405,4 +479,80 @@ mod tests {
         }
         assert_eq!(iter.next(), None);
     }
+
+    #[test]
+    fn test_delta() {
+        // The trickiness here almost entirely relates to negative maximum values for signed
+        // integer types
+        assert_eq!(delta::<u8, u8>(0, 2), 2);
+        assert_eq!(delta::<u8, u8>(0, u8::max_value()), u8::max_value());
+        assert_eq!(delta::<i8, u8>(-2, 0), 2);
+        assert_eq!(delta::<i8, u8>(-2, 2), 4);
+        assert_eq!(delta::<i8, u8>(i8::min_value(), 0), 128);
+        assert_eq!(delta::<i8, u8>(0, i8::max_value()), 127);
+        assert_eq!(
+            delta::<i8, u8>(i8::min_value(), i8::max_value()),
+            u8::max_value()
+        );
+        assert_eq!(delta::<i8, u8>(i8::min_value(), i8::min_value()), 0);
+        assert_eq!(
+            delta::<i32, u32>(i32::min_value(), i32::max_value()),
+            (i32::max_value() as u32) * 2 + 1
```

Comment:
[minor] Why not `u32::max_value()`? This seems to be testing the assumption that `(i32::max_value() as u32) * 2 + 1` == `u32::max_value()` (which should be true anyway).

Reply:
Yes, it's just a belt and braces thing, partly to remind people of the asymmetry.